model/project_parser.go

```diff
@@ -484,10 +484,20 @@ func (pss *parserStringSlice) UnmarshalYAML(unmarshal func(interface{}) error) e
 	return nil
 }
 
-// LoadProjectForVersion returns the project for a version, either from the parser project or the config string.
-// If read from the config string and shouldSave is set, the resulting parser project will be saved.
-func LoadProjectForVersion(v *Version, id string, shouldSave bool) (ProjectInfo, error) {
-	var pp *ParserProject
+// FindAndTranslateProjectForVersion translates a parser project for a version into a Project
+func FindAndTranslateProjectForVersion(v *Version, id string) (*Project, *ParserProject, error) {
+	pp, err := ParserProjectFindOneById(v.Id)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "finding parser project")
+	}
+	pp.Identifier = utility.ToStringPtr(id)
+	var p *Project
+	p, err = TranslateProject(pp)
```

Comment:
I think it's more correct to return nil for the project/parser project if there's an error